packages/now-build-utils/src/detect-builders.ts

```diff
@@ -173,19 +173,17 @@ export async function detectBuilders(
         src: 'public/**/*',
         config,
       });
-    } else if (builders.length > 0) {
-      // We can't use pattern matching, since `!(api)` and `!(api)/**/*`
-      // won't give the correct results
-      builders.push(
-        ...files
-          .filter(name => !name.startsWith('api/'))
-          .filter(name => !(name === 'package.json'))
-          .map(name => ({
-            use: '@now/static',
-            src: name,
-            config,
-          }))
-      );
+    } else if (
+      builders.length > 0 &&
+      files.some(f => !f.startsWith('api/') && f !== 'package.json')
+    ) {
+      // Everything besides the api directory
+      // and package.json can be served as static files
+      builders.push({
+        use: '@now/static',
+        src: '!api/**',
```

Comment:
Won't `package.json` be served as a static file with this?

Reply:
No, theres a test for that and you can see the line above filters out package.json

Reply:
> the line above filters out package.json
Isn't it just a condition and not a filter?

Reply:
(and I couldn't find the test 😅)

Reply:
Also perhaps based on the comment on line 181.

Reply:
Yes, I see now there is still a bug that the unit tests wont catch because they are not running the glob `src` string against the file inputs.

@TooTallNate It seems like I can't make a glob that negates two things. See this example: https://runkit.com/embed/ch57dy4bk27v

Reply:
Ok, this has been fixed thanks to @lucleray 👍